extensions/amp-story-shopping/0.1/test/test-amp-story-shopping-config.js

```diff
@@ -36,7 +56,49 @@ describes.realWin(
 
     it('should build shopping config component', () => {
       expect(() => shoppingConfig.layoutCallback()).to.not.throw();
-      expect(shoppingConfig.isLayoutSupported(Layout.NODISPLAY)).to.be.true;
+    });
+
+    it('throws on no config', async () => {
+      expectAsyncConsoleError(async () => {
+        expect(async () => {
+          await shoppingConfig.buildCallback();
+        }).to.throw(
+          /The amp-story-auto-ads:config should be inside a <script> tag with type=\"application\/json\"​​​/
+        );
+      });
+    });
+
+    it('does use remote config when src attribute is provided', async () => {
+      const exampleURL = 'foo.example';
+      element.setAttribute('src', exampleURL);
+
+      const expectedRemoteResult =
+        '{"city-pop":{"product-tag-id":"city-pop","brand-label":"...","brand-favicon":"...","product-title":"Plastic Love","product-price":"...","product-images":"...","product-details":"...","reviews-page":"...","reviews-data":"...","cta-text":1,"shipping-text":1},"k-pop":{"product-tag-id":"k-pop","brand-label":"...","brand-favicon":"...","product-title":"Gangnam Style","product-price":"...","product-images":"...","product-details":"...","reviews-page":"...","reviews-data":"...","cta-text":1,"shipping-text":1},"eurodance":{"product-tag-id":"eurodance","brand-label":"...","brand-favicon":"...","product-title":"Crystal King Battle","product-price":"...","product-images":"...","product-details":"...","reviews-page":"...","reviews-data":"...","cta-text":1,"shipping-text":1}}';
+
+      expect(
+        JSON.stringify(storeService.get(StateProperty.SHOPPING_STATE))
+      ).to.eql(expectedRemoteResult);
+    });
+
+    it('does use inline config when remote src is invalid', async () => {
+      const exampleURL = 'invalidRemoteURL';
+      element.setAttribute('src', exampleURL);
+
+      shoppingConfig.buildCallback().then(() => {
+        expect(shoppingConfig.getInlineConfig_).to.be.called();
+      });
+    });
+
+    it('Test Invalid Remote Config URL', async () => {
+      const exampleURL = 'invalidRemoteURL';
+      element.setAttribute('src', exampleURL);
+      expectAsyncConsoleError(async () => {
+        expect(async () => {
+          await shoppingConfig.buildCallback();
+        }).to.throw(
+          /'amp-story-auto-ads:config error determining if remote config is valid json: bad url or bad json'​​​/
+        );
+      });
```

Comment:
I wonder if we need these tests since we're using `loadConfigImpl`, which has tests already.

Reply:
I don't think having more unit tests will hurt. As we've seen before, loadConfigImpl didn't pass certain cases where the inline config was supposed to be used when the remote failed. And also as you mentioned-  unit tests don't add to the bundle size so it won't affect the runtime for users, hence I am in favor of keeping these tests.

Reply:
This is a matter of organization of our codebase. Not about bundle size.
This functionality is not part of this component. Conceptually they should be in `test-amp-story-request-service.js`.  

`loadConfigImpl didn't pass certain cases where the inline config was supposed to be used when the remote failed.`
New tests should be written in `test-amp-story-request-service.js` to cover this.